addons/l10n_eg_edi_eta_imp/models/eta_thumb_drive.py

```diff
@@ -0,0 +1,15 @@
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
+
+
+class EtaThumbDrive(models.Model):
+    _inherit = 'l10n_eg_edi.thumb.drive'
+
+    name = fields.Char()
+    active = fields.Boolean(default=True)
+
+    @api.constrains('active', 'user_id', 'company_id')
+    def _constrains_active_thumb(self):
+        for drive in self:
+            if self.search_count([('company_id', '=', drive.company_id.id), ('user_id', '=', drive.user_id.id)]) > 1:
+                raise ValidationError(_('You can only have one drive active at a time per user and per company'))
```

Comment:
What happens with the sql constraint in the original module?  Because I am not sure it will automatically update with the active in mind...  (because if it took it into account, this constraint would be unnecessary)